client/ts/src/client.ts

```diff
@@ -194,6 +203,172 @@ export class ManifestClient {
+    marketPk: PublicKey,
+    payerPub: PublicKey,
+  ): Promise<SetupData> {
+    const setupData: SetupData = {
+      setupNeeded: true,
+      instructions: [],
+      wrapperKeypair: null,
+    };
+    const existingWrappers: GetProgramAccountsResponse =
+      await connection.getProgramAccounts(WRAPPER_PROGRAM_ID, {
+        filters: [
+          // Dont check discriminant since there is only one type of account.
+          {
+            memcmp: {
+              offset: 8,
+              encoding: 'base58',
+              bytes: payerPub.toBase58(),
+            },
+          },
+        ],
+      });
+
+    if (existingWrappers.length == 0) {
+      const wrapperKeypair: Keypair = Keypair.generate();
+      setupData.wrapperKeypair = wrapperKeypair;
+
+      const createAccountIx: TransactionInstruction =
+        SystemProgram.createAccount({
+          fromPubkey: payerPub,
+          newAccountPubkey: wrapperKeypair.publicKey,
+          space: FIXED_WRAPPER_HEADER_SIZE,
+          lamports: await connection.getMinimumBalanceForRentExemption(
+            FIXED_WRAPPER_HEADER_SIZE,
+          ),
+          programId: WRAPPER_PROGRAM_ID,
+        });
+      setupData.instructions.push(createAccountIx);
+
+      const createWrapperIx: TransactionInstruction =
+        createCreateWrapperInstruction({
+          owner: payerPub,
+          payer: payerPub,
+          wrapperState: wrapperKeypair.publicKey,
+        });
+      setupData.instructions.push(createWrapperIx);
+
+      const claimSeatIx: TransactionInstruction = createClaimSeatInstruction({
+        manifestProgram: MANIFEST_PROGRAM_ID,
+        owner: payerPub,
+        market: marketPk,
+        payer: payerPub,
+        wrapperState: wrapperKeypair.publicKey,
+      });
+      setupData.instructions.push(claimSeatIx);
+
+      return setupData;
+    }
+
+    // Otherwise there is an existing wrapper
+    const wrapperResponse: Readonly<{
+      account: AccountInfo<Buffer>;
+      pubkey: PublicKey;
+    }> = existingWrappers[0];
+    const wrapperData: WrapperData = Wrapper.deserializeWrapperBuffer(
+      wrapperResponse.account.data,
+    );
+
+    const existingMarketInfos: MarketInfoParsed[] =
+      wrapperData.marketInfos.filter((marketInfo: MarketInfoParsed) => {
+        return marketInfo.market.toBase58() == marketPk.toBase58();
+      });
+    if (existingMarketInfos.length > 0) {
+      setupData.setupNeeded = false;
+      return setupData;
+    }
+
+    // There is a wrapper, but need to claim a seat.
+    const claimSeatIx: TransactionInstruction = createClaimSeatInstruction({
+      manifestProgram: MANIFEST_PROGRAM_ID,
+      owner: payerPub,
+      market: marketPk,
+      payer: payerPub,
+      wrapperState: wrapperResponse.pubkey,
+    });
+    setupData.instructions.push(claimSeatIx);
+
+    return setupData;
+  }
+
+  /**
+   * Create a new client. throws if setup ixs are needed. Call ManifestClient.getSetupIxs to check if ixs are needed.
```

Comment:
Comment that this is the one that UIs should be expected to use, other one is for typescript trading bots that have access to a keypair.